src/test/java/com/hazelcast/aws/AwsDiscoveryStrategyTest.java

```diff
@@ -135,4 +135,34 @@ public void discoverNodesOneAddressOnePort()
         assertEquals(new Address(publicAddress, port), node.getPublicAddress());
         assertFalse(iterator.hasNext());
     }
+
+    @Test
+    public void validateValidRegion() {
+        List<String> awsRegions = Arrays.asList(
+            "us-east-2", "us-west-2", "ap-south-1", "ap-northeast-3", "ap-southeast-2",
+            "ca-central-1", "cn-north-1", "cn-northwest-1", "us-gov-east-1");
+
+        awsRegions.forEach(region -> {
```

Comment:
Nit: I actually think iterating in test is a bad thing, because when the test fails, you don't really know which parameter was it failing on. So, I'd rather have really a list of method calls:

```
awsDiscoveryStrategy.validateRegion("us-east-2");
awsDiscoveryStrategy.validateRegion("us-west-2");
...
```

The other option is to use parameterized tests, but I think it's an overkill for this simple scenario.

Reply:
I agree with not iterating on test, I'd really rather have the Parameterized tests, as I said on my other comment.

This was a way of trying to cover as much ground as possible, but I'll focus on two passing and two failing scenarios.